plugins/FileUpload/class.fileupload.plugin.php

```diff
@@ -827,7 +827,11 @@ public function PostController_Upload_Create($Sender) {
          $this->EventArguments['OriginalFilename'] = $FileName;
          $Handled = FALSE;
          $this->EventArguments['Handled'] =& $Handled;
-         $this->FireAs('Gdn_Upload')->FireEvent('SaveAs');
+         if($ImageType !== FALSE) {
+            $this->FireAs('Gdn_Upload')->FireEvent('SaveImageAs');
```

Comment:
Not sure ... There's a good chance that at the time I thought I was fitting
in with how I thought it works. Thanks for catching that!
On Feb 22, 2015 3:41 PM, "Lincoln Russell" <notifications@github.com> wrote:

> In plugins/FileUpload/class.fileupload.plugin.php
> <https://github.com/vanilla/addons/pull/182#discussion_r25135597>:
>
> > @@ -827,7 +827,11 @@ public function PostController_Upload_Create($Sender) {
> >           $this->EventArguments['OriginalFilename'] = $FileName;
> >           $Handled = FALSE;
> >           $this->EventArguments['Handled'] =& $Handled;
> > -         $this->FireAs('Gdn_Upload')->FireEvent('SaveAs');
> > +         if($ImageType !== FALSE) {
> > +            $this->FireAs('Gdn_Upload')->FireEvent('SaveImageAs');
>
> See #189 <https://github.com/vanilla/addons/pull/189> - had to modify
> this.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/vanilla/addons/pull/182/files#r25135597>.
>